packages/orbit-components/src/tmp-tailwind/InputField/index.tsx

```diff
@@ -0,0 +1,318 @@
+    ariaExpanded,
+    ariaControls,
+    autoFocus,
+    spaceAfter,
+    id,
+    width = "100%",
+    inputMode,
+    insideInputGroup,
+    dataAttrs,
+    role,
+  }: Props = props;
+
+  const forID = useRandomId();
+  const inputId = id || forID;
+
+  const hasTooltip = Boolean(error || help);
+
+  const {
+    tooltipShown,
+    tooltipShownHover,
+    setTooltipShown,
+    setTooltipShownHover,
+    labelRef,
+    iconRef,
+    handleFocus,
+  } = useErrorTooltip<HTMLInputElement, HTMLDivElement>({ onFocus, hasTooltip });
+
+  const shown = tooltipShown || tooltipShownHover;
+  const fieldRef = React.useRef(null);
+  const Component = label ? "label" : "div";
+
+  return (
+    <Component
+      className={cx(
+        "font-base relative block flex-[1_1_100%]",
+        spaceAfter && spaceAfterClasses[spaceAfter],
+      )}
+      style={{ width }}
+      ref={fieldRef}
+      htmlFor={label ? inputId : undefined}
+      onMouseEnter={() => (disabled && inlineLabel ? setTooltipShownHover(true) : undefined)}
+      onMouseLeave={() => (disabled && inlineLabel ? setTooltipShownHover(false) : undefined)}
+    >
+      {label && !inlineLabel && (
+        <FormLabel
+          inlineLabel={inlineLabel}
+          filled={!!value}
+          required={required}
+          error={!!error}
+          help={!!help}
+          labelRef={labelRef}
+          iconRef={iconRef}
+          onMouseEnter={() => setTooltipShownHover(true)}
+          onMouseLeave={() => setTooltipShownHover(false)}
+        >
+          {label}
+        </FormLabel>
+      )}
+      <div
+        className={cx(
+          "orbit-input-field-input-container",
+          "flex flex-row items-center justify-between",
+          "relative box-border",
+          "h-form-box-normal text-form-element-normal",
+          disabled ? "cursor-not-allowed" : "cursor-text",
+          disabled || readOnly
+            ? "text-form-element-disabled-foreground"
+            : "text-form-element-filled-foreground",
+        )}
+        ref={label ? null : labelRef}
+      >
+        {inlineLabel && !tags && (error || help) ? (
+          <Prefix>
+            {help && !error && (
+              <div className="flex" ref={iconRef}>
+                <InformationCircle color="info" size="small" />
+              </div>
+            )}
+            {error && (
+              <div className="flex" ref={iconRef}>
+                <AlertCircle color="critical" size="small" />
+              </div>
+            )}
+          </Prefix>
+        ) : (
+          prefix && <Prefix>{prefix}</Prefix>
+        )}
+        {label && inlineLabel && (
+          <div
+            className={cx(
+              "orbit-input-field-inline-label",
+              "flex items-center justify-center",
+              "pointer-events-none h-full",
+              "[&>span]:mb-0",
+              "[&>span]:text-form-box-normal [&>span]:whitespace-nowrap [&>span]:leading-normal",
+              "[&>span]:z-[3]",
```

Comment:
It can be span only there. As it's from the label which is a span. 
Actually, we should avoid adding classes everywhere, as it opens pandora's box for overwriting our components

Reply:
I agree with overwriting styles. But ultimately this approach breaks the abstraction barrier on the FormLabel. If we decide to change the rendered element in `orbit-form-label` to a `p` or a `div` (it's unlikely, but for whatever reason it can happen, and it's just a general example here) then this selector will not have the effect we want it to have